include/boost/geometry/algorithms/detail/overlay/pointlike_pointlike.hpp

```diff
@@ -256,8 +259,9 @@ struct multipoint_multipoint_point
     {
         typedef geometry::less<void, -1, typename Strategy::cs_tag> less_type;
 
-        if ( OverlayType != overlay_difference
-             && boost::size(multipoint1) > boost::size(multipoint2) )
+        if (BOOST_GEOMETRY_CONDITION(
```

Comment:
What do you think about adding a macro allowing to conditionally use C++17 `if constexpr`?

Reply:
It can probably be integrated into `BOOST_GEOMETRY_CONDITION` ? Such that we don't need another one?

Reply:
@barendgehrels Unfortunately we do need two separate macros. AFAIR there are rare places where a condition can be constant or not, e.g. some policy can define a variable as static or dynamic member. Of course we could agree to drop static members in such cases.

Another thing is that I think we should consider something which looks more like the language feature. I'd also prefer to keep `if` so to not include it in the macro in order to allow the editor to highlight the key word to make it more readable. So maybe something like this:
```
if constexpr (OverlayType != overlay_difference)
vs
if BOOST_GEOMETRY_CONSTEXPR (OverlayType != overlay_difference)
```

Reply:
See the condition below in pj_transform.hpp line 255:
```
if( BOOST_GEOMETRY_CONDITION(srcdefn.vto_meter != 1.0 && dimension > 2) )
```
where `srcdefn.vto_meter` may or may not be static but `dimension` is always static. In this case `if constexpr` should be used for `dimension` and `BOOST_GEOMETRY_CONDITION` for `srcdefn.vto_meter`. Or `vto_meter` should always be non-static member, also in static projections .

Reply:
OK, I see, and saw it in your new PR this morning. Cool.